src/metabase/driver/sql_jdbc/actions.clj

```diff
@@ -15,32 +15,57 @@
                     :status-code status-code}
                    more-info))))
 
+(defn- check-one-row-effected [conn query raw-hsql] ;
+  (let [select-hsql     (-> raw-hsql (assoc :select [[:%count.* :row-count]]))
+        row-count       (:row_count (first (jdbc/query conn (hformat/format select-hsql))) 0)]
+    (when (not= 1 row-count)
```

Comment:
```suggestion
    (when-not (= row-count 1)
```

Another nit. This is a [Yoda Condition](https://en.wikipedia.org/wiki/Yoda_conditions)